### PR TITLE
Send users to the new sell tab when they tap on the artist consign CTA

### PR DESCRIPTION
**Pull request by pepopowitz**
Addresses https://artsyproduct.atlassian.net/browse/CSGN-210.

When we built out the new sell tab, we forgot that tapping the "consign with Artsy" button on an artist page would direct the user to the new landing page. It wasn't so much a problem that we were sending them there...but they were sent to the landing page in the context of a modal without a mechanism for closing or going back. They were dead-ended, and the only way to escape was to restart the app.

This is what it looked like:

artist-consign-flow-small-stuck

This PR updates the destination of that button to be the new sell tab (if the feature flag is enabled). 

This works really well when they tap an artist consign button while browsing within the home or search tabs of the app:

artist-consign-flow-small-good

It is a little less great when they tap an artist consign button while browsing within the new sell tab: 

artist-consign-flow-small-confusing

The main issue there being that when they're brought to the tab landing page, it has retained their previous scroll position, which feels confusing. 

We're in talks amongst the consignments team to figure out what to do about that last issue. For now, even though it may be confusing, it is significantly better than dead-ending them.

### CODE REVIEW

File: src/lib/Components/Artist/__tests__/ArtistConsignButton-tests.tsx [hunk 1]
```diff
@@ -137,6 +143,25 @@ describe("ArtistConsignButton", () => {
         destination_path: "/consign/submission",
       })
     })
+
+    it("tracks the sales tab destination if feature flag is enabled", () => {
```
Comment (damassi) [inline src/lib/Components/Artist/__tests__/ArtistConsignButton-tests.tsx:147]: 👍

File: src/lib/Components/Artist/__tests__/ArtistConsignButton-tests.tsx [hunk 2]
```diff
@@ -186,5 +211,24 @@ describe("ArtistConsignButton", () => {
...
```
Comment (damassi) [inline src/lib/Components/Artist/__tests__/ArtistConsignButton-tests.tsx:215]: Once this is deployed and feature flag removed will these tests have to be updated? If so can we add a TODO somewhere?

Comment (pepopowitz) [reply src/lib/Components/Artist/__tests__/ArtistConsignButton-tests.tsx:215]:
```suggestion
    // TODO: make this the default case once the feature flag is removed
    it("tracks the sales tab destination if feature flag is enabled", () => {
```

File: src/lib/Components/Artist/__tests__/ArtistConsignButton-tests.tsx (hunk 1 above)
Comment (pepopowitz) [inline src/lib/Components/Artist/__tests__/ArtistConsignButton-tests.tsx:147]:
```suggestion
    // TODO: make this the default case once the feature flag is removed
    it("tracks the sales tab destination if feature flag is enabled", () => {
```

**Comment by jonallured** [comment]
LGTM!! 🎉 #mergeongreen

**Pull request merged**